src/components/Executions/ExecutionFilters.tsx

```diff
@@ -53,13 +58,37 @@ const RenderFilter: React.FC<{ filter: FilterState }> = ({ filter }) => {
  * This allows for the consuming code to have direct access to the
  * current filters without relying on complicated callback arrangements
  */
-export const ExecutionFilters: React.FC<{ filters: FilterState[] }> = ({
-    filters
-}) => {
+export const ExecutionFilters: React.FC<{
+    filters: (FilterState | BooleanFilterState)[];
+}> = ({ filters }) => {
     const styles = useStyles();
+
     return (
         <div className={styles.container}>
-            {filters.map(filter => {
+            {filters.map((filter: any) => {
+                if (filter.hidden) {
+                    return null;
+                }
+                if (filter.type === 'boolean') {
+                    const handleChange = (
+                        event: React.ChangeEvent<HTMLInputElement>
+                    ) => filter.setActive(event.target.checked);
+
+                    return (
+                        <FormControlLabel
+                            key={filter.label}
+                            data-testid="checkbox"
```

Comment:
This is very generic. Test ids should be unique. But if we had multiple boolean filters being rendered, they would all have the same id.  Also, test id is supposed to be a last resort for getting a component when using react-testing-library, since it doesn't test what the user is seeing. We should be able to get this component by its label instead. [See here for more info](https://testing-library.com/docs/queries/about#priority)